hbase-client/src/main/java/org/apache/hadoop/hbase/util/PoolMap.java

```diff
@@ -85,18 +83,17 @@ public V put(K key, V value) {
   @SuppressWarnings("unchecked")
   @Override
   public V remove(Object key) {
-    Pool<V> pool = pools.remove(key);
-    if (pool != null) {
-      removeValue((K) key, pool.get());
-    }
+    pools.remove(key);
     return null;
   }
```

Comment:
I know it was already returning null before this PR, but for compliance with Map interface definition, we should rather return `pools.remove(key)`

Reply:
Removed Map interface.